paddle/operators/crop_op.cc

```diff
@@ -0,0 +1,87 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+#include "paddle/operators/crop_op.h"
+#include <boost/lexical_cast.hpp>
+
+namespace paddle {
+namespace operators {
+
+using framework::Tensor;
+
+class CropOp : public framework::OperatorWithKernel {
+ public:
+  using framework::OperatorWithKernel::OperatorWithKernel;
+
+ protected:
+  void InferShape(const framework::InferShapeContext &ctx) const override {
+    auto dim0 = ctx.Input<Tensor>("X")->dims();
+    auto Y = ctx.Input<Tensor>("Y");
+    if (Y == nullptr) {
+      auto shape = GetAttr<std::vector<int>>("shape");
+      PADDLE_ENFORCE_EQ(
+          shape.size(), dim0.size(),
+          "Shape size should be equal to dimention size of input tensor.");
+      std::vector<int64_t> tensor_shape(shape.size());
+      for (int i = 0; i < shape.size(); ++i) {
+        tensor_shape[i] = (int64_t)shape[i];
+      }
+      ctx.Output<Tensor>("Out")->Resize(
+          paddle::framework::make_ddim(tensor_shape));
+    } else {
+      ctx.Output<Tensor>("Out")->Resize(Y->dims());
+    }
+  }
+};
+
+class CropOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  CropOpMaker(framework::OpProto *proto, framework::OpAttrChecker *op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("X", "The input of crop op");
+    AddInput("Y", "The input used as reference for cropping. ");
+    AddOutput("Out", "The output of crop op.");
+    AddComment(R"DOC(
```

Comment:
This is a bad documentation, `The offsets for cropping` contains no information at all. Maybe describe some detail like (not necessary) 
1. in which scene this Op will be used.
2. Input, Output of the Op.
3. the detail of this attribute, such as `shape` refers to the area shape of interest in the source image.

Reply:
Thx. I will refine it.